libraries/ts-command-line/src/CommandLineParameter.ts

```diff
@@ -1,46 +1,82 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
-/**
- * @public
- */
-export interface IConverterFunction<T> {
-  (initial: any): T; /* tslint:disable-line:no-any */
-}
+import {
+  IBaseCommandLineDefinition,
+  ICommandLineFlagDefinition,
+  ICommandLineStringDefinition,
+  ICommandLineStringListDefinition,
+  ICommandLineIntegerDefinition,
+  ICommandLineChoiceDefinition,
+  IBaseCommandLineDefinitionWithArgument
+} from './CommandLineDefinition';
 
 /**
- * @internal
+ * Identifies the kind of a CommandLineParameter.
+ * @public
  */
-export interface ICommandLineParserData {
-  action: string;
-  [key: string]: any; /* tslint:disable-line:no-any */
+export enum CommandLineParameterKind {
+  /** Indicates a CommandLineChoiceParameter */
+  Choice,
+  /** Indicates a CommandLineFlagParameter */
+  Flag,
+  /** Indicates a CommandLineIntegerParameter */
+  Integer,
+  /** Indicates a CommandLineStringParameter */
+  String,
+  /** Indicates a CommandLineStringListParameter */
+  StringList
 }
 
 /**
  * The base class for the various command-line parameter types.
- *
- * @remarks
- *
- * The "subclasses" of this class are not actually constructed directly.  Instead,
- * they are used as shorthand for various parameterizations of CommandLineParameter<T>.
  * @public
  */
-export class CommandLineParameter<T> {
-  private _converter: IConverterFunction<T>;
+export abstract class CommandLineParameter<T> {
```

Comment:
><T> [](start = 42, length = 3)

We don't need this generic type anymore

Reply:
Good point - fixed

---
In reply to: [182192788](https://github.com/Microsoft/web-build-tools/pull/611#discussion_r182192788) [](ancestors = 182192788)